JsonLogic.Net/EvaluateOperators.cs

```diff
@@ -300,19 +300,37 @@ private Type DictionaryType(object d)
         {
             return d.GetType().GetTypeInfo().ImplementedInterfaces.FirstOrDefault(t => t.GetTypeInfo().IsGenericType && t.GetGenericTypeDefinition() == typeof(IDictionary<,>));
         }
-
-        // private Type EnumerableType(object d)
-        // {
-        //     return d.GetType().GetTypeInfo().ImplementedInterfaces.FirstOrDefault(t => t.IsGenericType && t.GetGenericTypeDefinition() == typeof(IEnumerable<>));
-        // }
-
-        private Func<IProcessJsonLogic, JToken[], object, object> DoubleArgsSatisfy(Func<double, double, bool> criteria)
+        
+        private Func<IProcessJsonLogic, JToken[], object, object> GenericArgsSatisfy(Func<Double, Double, bool> criteriaDouble, Func<string, string, bool> criteriaText)
         {
-            return (p, args, data) => {
-                var values = args.Select(a => a == null ? 0d : Double.Parse(p.Apply(a, data).ToString())).ToArray();
-                for (int i = 1; i < values.Length; i++) {
-                    if (!criteria(values[i-1], values[i])) return false;
+            return (p, args, data) =>
+            {
+                var values = args
+                    .Where(a => a != null)
+                    .Select(a => p.Apply(a, data))
+                    .Select(a => JToken.FromObject(a))
+                    .ToArray();
+                
+                // all values text?
+                var allText = values.All(a => a.Type == JTokenType.String);
+                if (allText)
+                {
+                    var valuesText = args.Select(a => a == null ? "" : p.Apply(a, data).ToString()).ToArray();
```

Comment:
the values are already generated. There is no need to re-apply the evaluations. You can use the values like `var textValues = values.Cast<strirng>().ToArray();`